File: src/main/java/org/example/bankramenserver/domain/report/domain/repository/MonthlyReportRepositoryCustom.java
package org.example.bankramenserver.domain.report.domain.repository;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public interface MonthlyReportRepositoryCustom {

    AmountSummaryRow findAmountSummary(
            UUID userId,
            LocalDate currentStartDate,
            LocalDate currentEndDate,
            LocalDate previousStartDate,
            LocalDate previousEndDate
    );

    List<CategoryExpenseRow> findCategoryExpenseComparisons(
            UUID userId,
            LocalDate currentStartDate,
            LocalDate currentEndDate,
            LocalDate previousStartDate,
            LocalDate previousEndDate
    );
}

File: src/main/kotlin/org/example/balogserver/domain/report/domain/repository/MonthlyReportRepositoryCustom.kt
package org.example.balogserver.domain.report.domain.repository

import java.time.LocalDate
import java.util.UUID

interface MonthlyReportRepositoryCustom {
    fun findAmountSummary(userId: UUID?, currentStartDate: LocalDate?, currentEndDate: LocalDate?, previousStartDate: LocalDate?, previousEndDate: LocalDate?): AmountSummaryRow
    fun findCategoryExpenseComparisons(userId: UUID?, currentStartDate: LocalDate?, currentEndDate: LocalDate?, previousStartDate: LocalDate?, previousEndDate: LocalDate?): List<CategoryExpenseRow>
}
